app/src/app/components/display-results/display-results.component.ts

```diff
@@ -31,8 +31,9 @@ export class DisplayResultsComponent {
   operationSet: OperationSet;
 
   constructor(operations: OperationService, router: Router) {
-    const results = operations.getResults();
-    results.valid ? this.operationSet = results : router.navigateByUrl(routes.home);
+    operations.results.subscribe((results) => {
+      results.valid ? this.operationSet = results : router.navigateByUrl(routes.home);
```

Comment:
This will probably be a bit easier to read if we use an if statement. The ternary is usually used to set a variable to one of two options rather than perform side effects.

Reply:
Updated 👍